### ISSUE TITLE
Slow startup of magisk app and startup logo is not visible 

### ISSUE THREAD

**Issue by RV7PR** [topjohnwu/Magisk#5662]
Device: Samsung Galaxy S20+ 5G
Android version: 12
Magisk version name: 704f9154
Magisk version code: 24305

When hiding magisk it asks for root permissions and then goes on and when i close the app and then reopen it starts very slow without showing the startup logo
[magisk_log_2022-03-30T13_51_37.log](https://github.com/topjohnwu/Magisk/files/8380221/magisk_log_2022-03-30T13_51_37.log)

https://user-images.githubusercontent.com/17615589/160832403-1cbb2944-9116-438d-8b80-2e4fa4b8a822.mp4

**Comment by rowhel95** [comment]
When using original apk it opens immediately with startup logo and with hidden apk it has a 2-3 second freeze without startup logo, thats normal?

**Comment by osm0sis** [comment]
The wait is only after the initial hiding.

**Comment by rowhel95** [comment]
It happens everytime when i open the app.
Rebooting doesnt help either. 
Just installed 24304 to test and it works.

**Comment by rowhel95** [comment]
Original apk:

https://user-images.githubusercontent.com/17615589/160928283-8e95a4a5-5078-4b05-b6d5-6f0aaf21f118.mp4

Hidden apk:

https://user-images.githubusercontent.com/17615589/160928352-7bd01be1-1963-44f2-983f-e8aa6647aec2.mp4

**Comment by Mouradmabrouk** [comment]
***@***.***

**Comment by osm0sis** [comment]
And show us 24304 on your device please

**Comment by rowhel95** [comment]
Hidden apk 24304:


https://user-images.githubusercontent.com/17615589/160935535-49f8a572-c27b-4ab6-bc1c-19dc27abe395.mp4

**Comment by rowhel95** [comment]
Do you need logcat for 24305 or 24304?

**Comment by rowhel95** [comment]
Logcat 24304:
[logcat_03-30-2022_23-55-43 (1).txt](https://github.com/topjohnwu/Magisk/files/8384505/logcat_03-30-2022_23-55-43.1.txt)

Logcat 24305:
[logcat_03-31-2022_00-19-04.txt](https://github.com/topjohnwu/Magisk/files/8384510/logcat_03-31-2022_00-19-04.txt)

**Comment by sn-o-w** [comment]
I see another issue in the first attached video.

Why Magisk app starts in English, then in Dutch after Magisk was hidden, and then eventually back to English after Magisk is forced closed and manually re-opened?! 🤔

**Comment by RV7PR** [comment]
Yeah its always the first time opening after hiding, also zygisk shows status as no and after reopen its yes again

**Comment by canyie** [comment]
Should be fixed in https://github.com/topjohnwu/Magisk/commit/cc79a96fa31c3c7b9d05177726465a0f126880dc

**Comment by RV7PR** [comment]
I can confirm that it's fixed! 👍🏼
